webapi/src/main/scala/org/knora/webapi/store/triplestore/api/TriplestoreServiceInMemory.scala

```diff
@@ -0,0 +1,363 @@
+import org.knora.webapi.messages.store.triplestoremessages.InsertGraphDataContentResponse
+import org.knora.webapi.messages.store.triplestoremessages.InsertTriplestoreContentACK
+import org.knora.webapi.messages.store.triplestoremessages.NamedGraphDataResponse
+import org.knora.webapi.messages.store.triplestoremessages.RdfDataObject
+import org.knora.webapi.messages.store.triplestoremessages.RepositoryUploadedResponse
+import org.knora.webapi.messages.store.triplestoremessages.ResetRepositoryContentACK
+import org.knora.webapi.messages.store.triplestoremessages.SparqlAskResponse
+import org.knora.webapi.messages.store.triplestoremessages.SparqlConstructRequest
+import org.knora.webapi.messages.store.triplestoremessages.SparqlConstructResponse
+import org.knora.webapi.messages.store.triplestoremessages.SparqlExtendedConstructRequest
+import org.knora.webapi.messages.store.triplestoremessages.SparqlExtendedConstructResponse
+import org.knora.webapi.messages.store.triplestoremessages.SparqlUpdateResponse
+import org.knora.webapi.messages.util.rdf.QuadFormat
+import org.knora.webapi.messages.util.rdf.RdfFeatureFactory
+import org.knora.webapi.messages.util.rdf.RdfFormatUtil
+import org.knora.webapi.messages.util.rdf.RdfInputStreamSource
+import org.knora.webapi.messages.util.rdf.RdfModel
+import org.knora.webapi.messages.util.rdf.RdfStringSource
+import org.knora.webapi.messages.util.rdf.SparqlSelectResult
+import org.knora.webapi.messages.util.rdf.SparqlSelectResultBody
+import org.knora.webapi.messages.util.rdf.SparqlSelectResultHeader
+import org.knora.webapi.messages.util.rdf.Statement
+import org.knora.webapi.messages.util.rdf.Turtle
+import org.knora.webapi.messages.util.rdf.VariableResultsRow
+import org.knora.webapi.store.triplestore.api.TriplestoreServiceInMemory.createEmptyDataset
+import org.knora.webapi.store.triplestore.defaults.DefaultRdfData
+import org.knora.webapi.store.triplestore.errors.TriplestoreException
+import org.knora.webapi.store.triplestore.errors.TriplestoreResponseException
+import org.knora.webapi.store.triplestore.errors.TriplestoreTimeoutException
+import org.knora.webapi.store.triplestore.errors.TriplestoreUnsupportedFeatureException
+import org.knora.webapi.util.ZScopedJavaIoStreams.byteArrayOutputStream
+import org.knora.webapi.util.ZScopedJavaIoStreams.fileInputStream
+import org.knora.webapi.util.ZScopedJavaIoStreams.outputStreamPipedToInputStream
+
+final case class TriplestoreServiceInMemory(datasetRef: Ref[Dataset], implicit val sf: StringFormatter)
+    extends TriplestoreService {
+  private val rdfFormatUtil: RdfFormatUtil = RdfFeatureFactory.getRdfFormatUtil()
+  override def doSimulateTimeout(): UIO[SparqlSelectResult] = ZIO.die(
+    TriplestoreTimeoutException(
+      "The triplestore took too long to process a request. This can happen because the triplestore needed too much time to search through the data that is currently in the triplestore. Query optimisation may help."
+    )
+  )
```

Comment:
not within the scope of this PR: instead of having a fleshed-out TriplestoreService implementation for simulating timeouts, wouldn't it make more sense to have a separate timeout-only implementation of the TriplestoreService?

Reply:
I have another commit which at least removes this useless method form the trait. There are other `simulateTimeout: Boolean` params which I am looking forward to remove as well.